internal/pkg/cache/singleton.go

```diff
@@ -27,9 +27,7 @@ type Singleton[V any] struct {
 	get   func(context.Context) (V, error)
 	value V
 	err   error
-	// Storing a bool to not deal with generic zero/nil comparisons.
-	called bool
-	lock   sync.RWMutex
+	once  sync.Once
```

Comment:
Unrelated change. The singleton can be written as a once.

Reply:
Undo this

Reply:
Reverted.